frontend/lint/eslint-rules/no-top-level-find-by-text.js

```diff
@@ -0,0 +1,43 @@
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function rule(context) {
+  return {
+    Identifier(node) {
+      // at the start of analyzing a code path
+      if (isFindByText(node) && hasParentTestBlock(node)) {
+        context.report({
+          node,
+          message:
+            "You should scope your findByText calls to a container using .within()",
+        });
+      }
+    },
+  };
+}
+
+const isTestBlock = node => {
+  return ["it", "before", "beforeEach"].includes(node?.callee?.name);
+};
+
+const hasParentTestBlock = node => {
+  // exactly 6 levels up tells us if it's a direct child of a test block
+  return isTestBlock(node.parent.parent.parent.parent.parent.parent);
```

Comment:
I don't know why but I just love this line so much

Reply:
sadly, this line had to die in favor of recursively traversing the AST. 😢